### PR TITLE
Fix for TCOTA-344

### PR DESCRIPTION
**Pull request by isururanawaka**

### CODE REVIEW

File: src/main/java/edu/iu/terracotta/service/app/impl/OutcomeServiceImpl.java [hunk 1]
```diff
@@ -176,33 +171,46 @@ public void updateOutcome(Long outcomeId, OutcomeDto outcomeDto) throws TitleVal
...
         Optional<Experiment> experiment = experimentService.findById(experimentId);
+        List<Assignment> assignmentList = allRepositories.assignmentRepository.findByExposure_Experiment_ExperimentId(experimentId);
         List<OutcomePotentialDto> outcomePotentialDtos = new ArrayList<>();
         if (experiment.isPresent()) {
             String canvasCourseId = StringUtils.substringBetween(experiment.get().getLtiContextEntity().getContext_memberships_url(), "courses/", "/names");
-            List<AssignmentExtended> assignmentExtendedList = canvasAPIClient.listAssignments(canvasCourseId,experiment.get().getPlatformDeployment());
-            for (AssignmentExtended assignmentExtended:assignmentExtendedList){
+            List<AssignmentExtended> assignmentExtendedList = canvasAPIClient.listAssignments(canvasCourseId, experiment.get().getPlatformDeployment());
+            for (AssignmentExtended assignmentExtended : assignmentExtendedList) {
+                List<Assignment> matched = assignmentList.stream().filter(x -> {
```
Comment (machristie) [inline src/main/java/edu/iu/terracotta/service/app/impl/OutcomeServiceImpl.java:197]: This should also filter out this experiment's consent assignment. You can get that lmsAssignmentId from `experiment.getConsentDocument().getLmsAssignmentId()`.

**Review by machristie — CHANGES_REQUESTED** [review]
Looking good @isururanawaka . See my one comment, this should also filter out this experiment's consent assignment. That lmsAssignmentId is stored on a different entity, ConsentDocument.

File: src/main/java/edu/iu/terracotta/service/app/impl/OutcomeServiceImpl.java (hunk 1 above)
Comment (isururanawaka) [reply src/main/java/edu/iu/terracotta/service/app/impl/OutcomeServiceImpl.java:197]: @machristie this is fixed

File: src/main/java/edu/iu/terracotta/service/caliper/impl/CaliperServiceImpl.java [hunk 2]
```diff
@@ -117,6 +117,21 @@ public void sendAssignmentStarted(Submission submission, SecuredInfo securedInfo
...
         String uuid = "urn:uuid:" + UUID.randomUUID();
+
+        Map<String, Object> extenstions = new HashMap<>();
+
+        Long assignmentId = submission.getAssessment().getTreatment().getAssignment().getAssignmentId();
+        Long assessmentId = submission.getAssessment().getAssessmentId();
+        Long treatmentId = submission.getAssessment().getTreatment().getTreatmentId();
+
+        extenstions.put("terracotta_assignment_id",assignmentId);
+        extenstions.put("terracotta_submission_id",submission.getSubmissionId());
```
Comment (machristie) [inline src/main/java/edu/iu/terracotta/service/caliper/impl/CaliperServiceImpl.java:128]: submission id is duplicated. Also can we factor out this logic?

File: src/main/java/edu/iu/terracotta/service/caliper/impl/CaliperServiceImpl.java [hunk 3]
```diff
@@ -117,6 +117,21 @@ public void sendAssignmentStarted(Submission submission, SecuredInfo securedInfo
...
+        extenstions.put("terracotta_submission_id",submission.getSubmissionId());
```
Comment (machristie) [inline src/main/java/edu/iu/terracotta/service/caliper/impl/CaliperServiceImpl.java:129]: Looking at TCOTA-443, I don't see a requirement to add submission id.But condition id and exposure id are listed in the issue but missing here.

File: src/main/java/edu/iu/terracotta/service/caliper/impl/CaliperServiceImpl.java [hunk 4]
```diff
@@ -117,6 +117,21 @@ public void sendAssignmentStarted(Submission submission, SecuredInfo securedInfo
...
+        extenstions.put("terracotta_assessmentId_id",assessmentId);
```
Comment (machristie) [inline src/main/java/edu/iu/terracotta/service/caliper/impl/CaliperServiceImpl.java:130]: I don't think we need assessment id, but if we do can we call it "terracotta_assessment_id"

**Review by machristie — CHANGES_REQUESTED** [review]
Reviewing TCOTA-433 changes.

**Pull request closed without merging**